drivers/clock_control/nrf_clock_calibration.c

```diff
@@ -200,139 +187,86 @@ static int get_temperature(s16_t *tvp)
 static void measure_temperature(struct k_work *work)
 {
 	s16_t temperature = 0;
-	s16_t diff;
+	s16_t diff = 0;
 	bool started = false;
-	int key;
 	int rc;
 
 	rc = get_temperature(&temperature);
 
-	key = irq_lock();
-
 	if (rc != 0) {
 		/* Temperature read failed: retry later */
-		to_idle();
+		start_cycle();
 		goto out;
 	}
 
 	diff = abs(temperature - prev_temperature);
 
-	if (cal_state != CAL_OFF) {
-		if ((calib_skip_cnt == 0) ||
-		    (diff >= CONFIG_CLOCK_CONTROL_NRF_CALIBRATION_TEMP_DIFF)) {
-			prev_temperature = temperature;
-			start_calibration();
-			started = true;
-		} else {
-			to_idle();
-			calib_skip_cnt--;
-			total_skips_cnt++;
-		}
+	if ((calib_skip_cnt == 0) ||
+		(diff >= CONFIG_CLOCK_CONTROL_NRF_CALIBRATION_TEMP_DIFF)) {
+		prev_temperature = temperature;
+		started = true;
+		start_hw_cal();
+	} else {
+		start_cycle();
```

Comment:
`start_cycle()` releases the lock under which this is running, right before the calibration count data is updated.  I would rather see `start_cycle()` be invoked after the state is updated.  Otherwise a maintainer could be concerned about a race condition ro simultaneous writes if something slips in and starts a new calibration process right away.

Reply:
moved `start_cycle()` lower